Blazorade.Bootstrap.Components/ProgressBar.razor.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Microsoft.AspNetCore.Components;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Blazorade.Bootstrap.Components
+{
+    public partial class ProgressBar
+    {
+        /// <summary>
+        /// If set to true, this progress bar will have a striped texture.
+        /// </summary>
+        [Parameter]
+        public bool IsStriped { get; set; }
+
+        /// <summary>
+        /// If set to true, any stripe will be animated.
+        /// </summary>
+        [Parameter]
+        public bool IsAnimated { get; set; }
+
+        /// <summary>
+        /// The minimum value represented by the progress bar. This must be less than both Value and ValueMax.
+        /// </summary>
+        [Parameter]
+        public double ValueMin { get; set; }
+
+        /// <summary>
+        /// The maximum value represented by the progress bar. This must be greater than both ValueMin and Value.
+        /// </summary>
+        [Parameter]
+        public double ValueMax { get; set; }
+
+        /// <summary>
+        /// The current value represented by the progress bar. This must fall between ValueMin and ValueMax.
+        /// </summary>
+        [Parameter]
+        public double Value { get; set; }
+
+        protected string CalculatedWidth {
+            get
+            {
+                return $"{Math.Round(Value / ValueMax, 2) * 100}%;";
+            }
+        }
+
+        protected override void OnParametersSet()
+        {
+            this.AddClasses(ClassNames.ProgressBars.ProgressBar);
+            this.AddAttribute("role", "progressbar");
+
+            if(this.IsStriped)
+            {
+                this.AddClasses(ClassNames.ProgressBars.Striped);
+            }
+
+            if (this.IsAnimated)
+            {
+                this.AddClasses(ClassNames.ProgressBars.Animated);
+            }
+
+            if (null != this.Color)
+            {
+                this.AddClasses(this.GetColorClassName("bg", this.Color));
+            }
+
+            // check and throw if contraints are busted
+            if (ValueMax < ValueMin)
+            {
+                throw new ArgumentOutOfRangeException("ValueMax must be greater than ValueMin");
+            }
+            else if (Value > ValueMax || Value < ValueMin)
+            {
+                throw new ArgumentOutOfRangeException("Value must be greater than ValueMin and less than ValueMax");
+            }
```

Comment:
This is probably more of an "academic" question, but does this add any value to a developer? What happens in Bootstrap, if you set the min value greater than max value, or if the actual value is less than the min value or greater than the max value?

Would it be more appropriate to use JavaScript interop and log a warning to the JS console? Developers tend to sit with the console open anyway.

Reply:
Perhaps. I wavered back and forth on this one, actually, wondering if this was value-added or not. It's certainly a part of the architecture of Blazor (and ASP.Net Core) that exceptions can be thrown at render-time, but I don't know if it's truly valuable to do so in this case.

Your suggestion about logging an exception to the console isn't a bad idea. Do we have a standard pattern for that elsewhere in the project?

Reply:
Would it be better to add a custom js function to do a console.warn(), or reference an injected ILogger, and then [set up a browser console logger as logger middleware](https://github.com/BlazorExtensions/Logging)? 

The latter depends on the user to set up that middleware (whatever their logging methodology or target may be) to get logs, but it's the standard logging pattern in ASP.Net Core, and I suspect Microsoft will want people to continue following that pattern in Blazor.

Reply:
I actually like that latter suggestion a lot better. I'm going to go ahead and implement it and you can decide whether you like it.

Reply:
Eek, looks like that package is broken on server-side Blazor, and seems to cause issues with WASM as well. Hrm. I like the idea of using an injected ILogger, so I think I'll keep that.

Reply:
Ok, the ILogger works fine server-side. Client-side they need to come up with a way to log to the browser console that conforms to this pattern, IMO. The package I linked would do it, but they need to fix their issues with circular dependencies first.

![image](https://user-images.githubusercontent.com/7681055/73962654-d1516980-48dc-11ea-883d-a2d7c92db00e.png)